model-optimizer/mo/utils/graph.py

```diff
@@ -306,3 +313,46 @@ def clear_tensor_names_info(nodes: list):
                     new_fw_info.append((fw_info[0], fw_info[1], None))
             set_edge_attribute_between_nodes(node, out_node, 'fw_tensor_debug_info', new_fw_info)
 
+
+def send_op_names_info(framework: str, graph: Graph):
+    """
+    This function sends information about operations in model.
+    :param framework: framework name.
+    :param graph: model graph.
+    """
+    op_counter = Counter()
+
+    def gather_op_statistics(g: Graph, op_c: Counter = op_counter):
+        op_c += g.op_names_statistic
+
+    for_graph_and_each_sub_graph_recursively(graph, gather_op_statistics)
+
+    t = tm.Telemetry()
+    for op_name in op_counter:
+        t.send_event('mo', 'op_count', "{}_{}".format(framework, op_name), op_counter[op_name])
+
+
+def send_shapes_info(framework: str, graph: Graph):
+    """
+    This function sends information about model input shapes.
+    :param framework: framework name.
+    :param graph: model graph.
+    """
+    shapes = []
+    for node in graph.get_op_nodes():
+        op_type = node.soft_get('type', None)
+        if op_type == 'Parameter':
+            if 'shape' in node:
+                shapes.append(node['shape'])
+    t = tm.Telemetry()
+
+    if shapes:
+        shape_str = ""
+        is_dynamic = "0"
+        for shape in shapes:
+            shape_str += np.array2string(shape) + ","
+            if not all(shape > 0):
+                is_dynamic = "1"
```

Comment:
@lazarevevgeny, @popovaan, is it implemented now that non-positive values are used for dynamic dimensions? I heard that we insert just `1` for dynamic dimension for further broadcasting.

Reply:
This function checks Parameter nodes only. The MO when reads the model reads dynamic values as -1 (TF) or 0 (ONNX). So the case when we insert `1` is not applicable here.

Reply:
I got it. It seems to be not about dynamism.

Reply:
Roman, you are actually right. The better approach is to rename this events to "partially_defined_shape" instead of "dynamic_input_shape"